api/dps/index.go

```diff
@@ -45,7 +45,7 @@ func IndexFromAPI(client APIClient) *Index {
 	return &i
 }
 
-// First returns the height of the first finalized block that was indexed.
+// First returns the height of the first indexed finalized block.
```

Comment:
I don't like this change, either. `finalized block` is a term within the Flow system. `indexed block` is not. By mixing them up like this, it becomes imo less clear.